bundles/org.openhab.binding.mffan/src/main/java/org/openhab/binding/mffan/internal/handler/MfFanHandler.java

```diff
@@ -0,0 +1,211 @@
+public class MfFanHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MfFanHandler.class);
+
+    private @Nullable MfFanConfiguration config;
+    private @Nullable FanRestApi api;
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private ExecutorService executor;
+
+    public MfFanHandler(Thing thing) {
+        super(thing);
+        this.executor = Executors.newSingleThreadExecutor();
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void initialize() {
+        this.logger.debug("Initializing MfFan handler '{}'", getThing().getUID());
+        this.config = getConfigAs(MfFanConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            this.api = new FanRestApi(this.config.getIpAddress().trim());
+            handleCommand(RefreshType.REFRESH);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        int pollingPeriod = (this.config.getPollingPeriod() == null) ? 120 : this.config.getPollingPeriod();
+        if (pollingPeriod > 0) {
+            // this.pollingJob = this.scheduler.scheduleWithFixedDelay(() -> update(this.api.getShadowBuffer()), 0,
+            // pollingPeriod, TimeUnit.SECONDS);
+            this.pollingJob = this.scheduler.scheduleWithFixedDelay(() -> handleCommand(RefreshType.REFRESH), 0,
+                    pollingPeriod, TimeUnit.SECONDS);
+
+            this.logger.debug("Polling job scheduled to run every {} sec. for '{}'", pollingPeriod,
+                    getThing().getUID());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        this.logger.debug("Disposing MF fan handler '{}'", getThing().getUID());
+
+        //
+        // Cancel the polling job.
+        //
+        if (this.pollingJob != null) {
+            this.pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+
+        //
+        // Shutdown the executor.
+        //
+        this.executor.shutdown();
+        try {
+            this.executor.awaitTermination(2, TimeUnit.SECONDS);
+        } catch (@SuppressWarnings("unused") InterruptedException e) {
+            this.executor.shutdownNow();
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            this.executor.execute(new CommandHandlerRunnable(channelUID, command));
+        } catch (@SuppressWarnings("unused") Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    String.format("Could not control device at IP address %s", this.config.getIpAddress()));
+        }
+    }
+
+    private void handleCommand(Command command) {
+        handleCommand(new ChannelUID("Null:Null:Null:Null"), command);
+    }
+
+    private final class CommandHandlerRunnable implements Runnable {
```

Comment:
Might be valid, but this is not a common pattern i see in most bindings, can you explain a bit on why it is moddled this way and not like many other bindings handling the commands inside the handdler class?